### DISCUSSION TITLE
Using private bitbucket-server repo tags as datasource

### DISCUSSION THREAD

**Discussion by samithShetty** [renovatebot/renovate#10576] [Request Help] [answered]
Hi, 

I'm currently trying to use self-hosted instance of renovate v25.49 in our bitbucket-server to check/update Jenkinsfiles for a specific module (which exists as a repo in the same bitbucket server, although in a different project). I want to check that repo and use it's tags as a datasource, is this possible? 

I am currently using the Regex Manager, which detects the version just fine, and I am trying to use the `bitbucket-tags` datasource, but I can't seem to find any information on them in the documentation and I'm not quite sure where to go from there... with the current setup, it is just giving a `Failed to look up dependency pipeline-utilities` message, and it doesn't seem to be even checking bitbucket, how can I configure it to do so?

Here's a snippet including the Regex manager in my config.js:
```
module.exports = {
    ...
    regexManagers: [
        {
            fileMatch: ['^Jenkinsfile$'],
            matchStrings: ["library 'pipeline-utilities@v(?<currentValue>.*?)'"],
            depNameTemplate: "pipeline-utilities",
            datasourceTemplate: "bitbucket-tags"
        }
    ]
}
```

And this is the line I'm grabbing from the Jenkinsfile :
```
library 'pipeline-utilities@v1.19.1'
```

Let me know what else I can provide to make it easier to help, thanks!

**Comment by viceice** [comment]
Not yet supported. The `bitbucket-tags` only works on cloud. You can try to play with `git-tags` datasource.

**Reply by samithShetty** [reply]
Got it, thanks. Is there any information on `git-tags` anywhere in the documentation? I only see it listed as a datasource, but nothing on how it actually works. How would I tell it where to find the repo in the case of that specific regex manager?

**Reply by viceice** [reply]
You need a hostrule for auth and assign git-tags as the datasource. Should be the same as assign bitbucket-tags datasource.

**Reply by samithShetty** [reply]
I haven't set up a hostRule before, would it be like this? Or should the matchHost be the full url for the project the dependency is in?
```
    hostRules: [
        {
            hostType: "git-tags",
            matchHost: "https://bitbucket.int.<company>.com",
            username: "<username>",
            password: "<password>",
            token: "<token>"
        }
    ]
```
Edit: maybe worth mentioning that the access key and account given to the renovate to put in these PRs already has read access to this repo. Do I need to give it credentials again? 
Also in either case of the url (whether giving the base  url or full url to project) I now get a ` fatal: 'pipeline-utilities' does not appear to be a git repository\nfatal: Could not read from remote repository.\n\nPlease make sure you have the correct access rights\nand the repository exists.\n` message, and I am pretty sure that all of the access rights are good. Maybe I should be specifying an entirely different host? or is this not where it is getting the url to search for the repo?

**Comment by samithShetty** [comment, marked as answer]
@viceice's suggestion of using `git-tags` worked! As for passing repo address, I just needed to add a lookupNameTemplate to the regexManager. Here are snippets for anybody else trying to do something similar:

_**config.js**_
```
module.exports = {
    ...
    regexManagers: [
        {
            fileMatch: ['^Jenkinsfile$'],
            matchStrings: ["library 'pipeline-utilities@v(?<currentValue>.*?)'"],
            depNameTemplate: "pipeline-utilities",
            datasourceTemplate: "git-tags",
            lookupNameTemplate: "https://bitbucket.int.<company>.com/project/pipeline-utilities.git" //Module Repo URL
        }
    ],
    hostRules: [
        {     
            hostType: "git-tags",
            matchHost: "https://bitbucket.int.<company>.com",
            username: "<username>",
            password: "<token>"
        }
    ]
}
```
Thanks for all of your help!
